MWI: Simpler auto-generated `tbot` service names

boxofrad opened the pull request:
Today, `tbot`'s automatically generated services names typically contain the full destination path, or other parameters, making them (A) not URL-safe and (B) ugly in the Web UI.

With this change, services will be named using a simpler scheme: `<type>-<count>`

For example:

- `application-tunnel-1`
- `kubernetes-argocd-1`

The counter is unique per service type, so the user could see `database-output-5` and go look at the fifth `database-output` service. It also treats services with user-chosen names as part of the sequence, so given this configuration:

```yaml
services:
  - type: application-output

  - type: application-output
    name: my-application

  - type: application-output
```

The service names would be `appliation-output-1`, `my-application`, and `application-output-3`.

changelog: MWI: `tbot`'s auto-generated service names are now simpler and easier to use in the `/readyz` endpoint

boxofrad committed:
MWI: Simpler auto-generated `tbot` service names

Today, `tbot`'s automatically generated services names typically contain the
full destination path, or other parameters, making them (A) not URL-safe and
(B) ugly in the Web UI.

With this change, services will be named using a simpler scheme: `<type>-<count>`

For example:

- `application-tunnel-1`
- `kubernetes-argocd-1`

The counter is unique per service type, so the user could see `database-output-5`
and go look at the fifth `database-output` service. It also treats services with
user-chosen names as part of the sequence, so given this configuration:

```yaml
services:
  - type: application-output

  - type: application-output
    name: my-application

  - type: application-output
```

The service names would be `appliation-output-1`, `my-application`, and
`application-output-3`.

changelog: MWI: `tbot`'s auto-generated service names are now simpler and easier to use in the `/readyz` endpoint

strideynet reviewed (approved):
Do we plan on backporting this?

boxofrad committed:
Update `/readyz` docs

boxofrad committed:
Use a `set.Set` instead of a map

boxofrad committed:
Remove old name validation test

boxofrad committed:
Fix `TestInitSymlink`

This was failing because calling `CheckAndSetDefaults` on the `BotConfig` was
copying pointers from `Outputs` into `Services`, and then we would marshal and
unmarshal the config to YAML, including "both" entries with the same generated
service name.

boxofrad committed:
Fix import ordering
